core/src/test/java/org/elasticsearch/indices/warmer/SimpleIndicesWarmerIT.java

```diff
@@ -173,6 +173,16 @@ public void deleteNonExistentIndexWarmerTest() {
         }
     }
 
+    @Test // issue 8991
+    public void deleteAllIndexWarmerDoesNotThrowWhenNoWarmers() {
+        createIndex("test");
+        try {
+            client().admin().indices().prepareDeleteWarmer().setIndices("test").setNames("_all").execute().actionGet();
```

Comment:
you can remove the try and catch here, if the test throws exception it fails anyway. maybe just do some sanity check on the response. Also maybe randomly add some other index name so `_all` is not always the only one.